src/uu/ls/src/ls.rs

```diff
@@ -1909,9 +2076,9 @@ fn display_size_or_rdev(metadata: &Metadata, config: &Config) -> String {
         let ft = metadata.file_type();
         if ft.is_char_device() || ft.is_block_device() {
             let dev: u64 = metadata.rdev();
-            let major = (dev >> 8) as u8;
-            let minor = dev as u8;
-            return format!("{}, {}", major, minor);
+            let major = (dev >> 24) as u8;
+            let minor = (dev & 0xff) as u8;
+            return format!("{},{:>5}", major, minor,);
```

Comment:
Ah this happened before :). See the discussion here: https://github.com/uutils/coreutils/issues/2140. Please revert this for now. You can try a full solution in another PR.

Reply:
FYI, here is what I've got to work on MacOS and Linux, but you're right that padding is still not correct.

```
#[cfg(any(target_os = "macos", target_os = "ios"))]
    {
        let ft = metadata.file_type();
        if ft.is_char_device() || ft.is_block_device() {
            let dev: u64 = metadata.rdev();
            let major = (dev >> 4) as u8;
            let minor = (dev & 0xff) as u8;
            return format!("{},{:>5}", major, minor,);
        }
    }

    #[cfg(target_os = "linux")]    
    {
        let ft = metadata.file_type();
        if ft.is_char_device() || ft.is_block_device() {
            let dev: u64 = metadata.rdev();
            let major = (dev >> 24) as u8;
            let minor = (dev & 0xff) as u8;
            return format!("{},{:>5}", major, minor,);
        }
    }
```

Reply:
That does make sense! It think it makes sense to extract this to a function with `cfg` attributes. But because we need to make it work for all platforms and change some tests, I think a separate PR is appropriate. Don't hesitate to open one!

Reply:
You were right.  This some how remained in.  During a switch of branches maybe?  It's out now.  Thanks.